### PR TITLE
matplotlib: 2.1.0

### PR DESCRIPTION
**Pull request by davidchall**
### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### Updates to existing formula: have you

- [x] Removed the `revision` line, if any, when bumping the version number?
- [x] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [x] Not altered the `bottle` section?
- [x] Checked that the tests still pass?

### CODE REVIEW

**Comment by iMichka** [comment]
Fails to build on High-Sierra, but may be related to numpy

```
In file included from src/ft2font_wrapper.cpp:5:
In file included from src/numpy_cpp.h:19:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/complex:246:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/cmath:313:9: error: no member named 'signbit' in the global namespace
using ::signbit;
      ~~^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/cmath:314:9: error: no member named 'fpclassify' in the global namespace
using ::fpclassify;
      ~~^
```

**Comment by davidchall** [comment]
I cannot reproduce this compiler error on my own machine. The main difference I noticed is that I do not have Xcode 9 installed (I use CLT instead).

BTW, `src/numpy_cpp.h` is a matplotlib source file, so I don't think this is related to the numpy installation.

**Comment by davidchall** [comment]
I confirm that these errors also appear when installing matplotlib 2.0.2 (the previous version) on macOS 10.12 (Sierra) with Xcode 9 installed. Here's a summary of results:

| Machine        | OS           | Toolchain  | Matplotlib version | Success? |
| :------------- |:------------- | :----- | :-----------------| :--------- |
| Jenkins      | 10.12, 10.13 | Xcode 9 | 2.1.0 | No
| My laptop      | 10.13      | CLT 9 | 2.1.0 | Yes
| My desktop | 10.12      |  Xcode 9 | 2.0.2 | No

In conclusion, this seems to be a problem with building against Xcode 9 and is not introduced by matplotlib 2.1.0.

When building against Xcode 9, it looks at 
`/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/cmath`
and when building against CLT9, it looks at
`/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/4.2.1/cmath`

**Comment by davidchall** [comment]
And here is a discussion of this very issue: https://github.com/Homebrew/homebrew-core/pull/18836

@ilovezfs I think SDKROOT is also relevant here.

**Comment by charleshan5330** [comment]
I run "brew install matplotlib"

$ brew install matplotlib
Updating Homebrew...
==> Installing matplotlib from homebrew/science
==> Downloading https://files.pythonhosted.org/packages/f5/f0/9da3ef24ea7eb0ccd12430a261b66eca36b924aeef06e17147f9f9d7d31
Already downloaded: /Users/hanliang/Library/Caches/Homebrew/matplotlib-2.0.2.tar.gz
==> Downloading https://files.pythonhosted.org/packages/dc/8c/7c9869454bdc53e72fb87ace63eac39336879eef6f2bf96e946edbf03e9
Already downloaded: /Users/hanliang/Library/Caches/Homebrew/matplotlib--setuptools-33.1.1.zip
==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/matplotlib/2.0.2/libexec --single-ver
==> Downloading https://files.pythonhosted.org/packages/c2/4b/137dea450d6e1e3d474e1d873cd1d4f7d3beed7e0dc973b06e8e10d3248
Already downloaded: /Users/hanliang/Library/Caches/Homebrew/matplotlib--Cycler-0.10.0.tar.gz
==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/matplotlib/2.0.2/libexec --single-ver
==> Downloading https://files.pythonhosted.org/packages/c5/60/6ac26ad05857c601308d8fb9e87fa36d0ebf889423f47c3502ef034365d
Already downloaded: /Users/hanliang/Library/Caches/Homebrew/matplotlib--functools32-3.2.3-2.tar.gz
==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/matplotlib/2.0.2/libexec --single-ver
==> Downloading https://files.pythonhosted.org/packages/38/bb/bf325351dd8ab6eb3c3b7c07c3978f38b2103e2ab48d59726916907cd6f
Already downloaded: /Users/hanliang/Library/Caches/Homebrew/matplotlib--pyparsing-2.1.10.tar.gz
==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/matplotlib/2.0.2/libexec --single-ver
==> Downloading https://files.pythonhosted.org/packages/51/fc/39a3fbde6864942e8bb24c93663734b74e281b984d1b8c4f95d64b0c21f
Already downloaded: /Users/hanliang/Library/Caches/Homebrew/matplotlib--python-dateutil-2.6.0.tar.gz
==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/matplotlib/2.0.2/libexec --single-ver
==> Downloading https://files.pythonhosted.org/packages/d0/e1/aca6ef73a7bd322a7fc73fd99631ee3454d4fc67dc2bee463e2adf6bb3d
Already downloaded: /Users/hanliang/Library/Caches/Homebrew/matplotlib--pytz-2016.10.tar.bz2
==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/matplotlib/2.0.2/libexec --single-ver
==> Downloading https://files.pythonhosted.org/packages/b3/b2/238e2590826bfdd113244a40d9d3eb26918bd798fc187e2360a8367068d
Already downloaded: /Users/hanliang/Library/Caches/Homebrew/matplotlib--six-1.10.0.tar.gz
==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/matplotlib/2.0.2/libexec --single-ver
==> Downloading https://files.pythonhosted.org/packages/b8/2f/49e53b0d0e94611a2dc624a1ad24d41b6d94d0f1b0a078443407ea2214c
Already downloaded: /Users/hanliang/Library/Caches/Homebrew/matplotlib--subprocess32-3.2.7.tar.gz
==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/matplotlib/2.0.2/libexec --single-ver
==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/matplotlib/2.0.2 --single-version-ext
Last 15 lines from /Users/hanliang/Library/Logs/Homebrew/matplotlib/09.python:
      ~~^
/Library/Developer/CommandLineTools/usr/include/c++/v1/cmath:322:9: error: no member named 'islessequal' in the global namespace
using ::islessequal;
      ~~^
/Library/Developer/CommandLineTools/usr/include/c++/v1/cmath:323:9: error: no member named 'islessgreater' in the global namespace
using ::islessgreater;
      ~~^
/Library/Developer/CommandLineTools/usr/include/c++/v1/cmath:324:9: error: no member named 'isunordered' in the global namespace
using ::isunordered;
      ~~^
/Library/Developer/CommandLineTools/usr/include/c++/v1/cmath:325:9: error: no member named 'isunordered' in the global namespace
using ::isunordered;
      ~~^
13 errors generated.
error: command 'clang' failed with exit status 1

READ THIS: https://docs.brew.sh/Troubleshooting.html

These open issues may also help:
matplotlib: 2.1.0 https://github.com/Homebrew/homebrew-science/pull/6382
Matplotlib –with-python3 doesn’t compile (because of qhull) https://github.com/Homebrew/homebrew-science/issues/5175
graph-tool dependencies and headers missing during installation https://github.com/Homebrew/homebrew-science/issues/5583
gubbins: influences on quast https://github.com/Homebrew/homebrew-science/issues/5752
rampart: Build a bottle for Linuxbrew https://github.com/Homebrew/homebrew-science/pull/5414
nest: revision for gsl https://github.com/Homebrew/homebrew-science/pull/5888
graph-tool failed to build on macOS 10.13 and Xcode 9.0 https://github.com/Homebrew/homebrew-science/issues/6394
Formula affected by Homebrew/brew#2482 https://github.com/Homebrew/homebrew-science/issues/5470
nest: Enable concurrent Python 2 & 3 support https://github.com/Homebrew/homebrew-science/pull/5620
Migrate most notable formulae to core https://github.com/Homebrew/homebrew-science/issues/6331
gubbins - Failed to import the site module (since python 3.4 -> 3.5) https://github.com/Homebrew/homebrew-science/issues/3490
graph-tool: symbol binding failed https://github.com/Homebrew/homebrew-science/issues/3259

**Comment by davidchall** [comment]
@iMichka I think this is ready to merge. Please let me know if you see any problems.

**Comment by davidchall** [comment]
Bump

**Comment by iMichka** [comment]
@BrewTestBot test this please

**Comment by davidchall** [comment]
Bump

**Comment by sjackman** [comment]
@BrewTestBot test this please

**Comment by sjackman** [comment]
```
==> brew linkage --test homebrew/science/matplotlib
Missing libraries:
  libgcc_s.so.1
  libstdc++.so.6
No unwanted system libraries
==> FAILED
```
https://travis-ci.org/Homebrew/homebrew-science/builds/287223814#L4605

`matplotlib` appears to depend on `gcc` now. Do you know why this change?

The current bottle does not depend on `gcc`:
```
❯❯❯ brew install matplotlib
❯❯❯ brew linkage matplotlib
System libraries:
…
Homebrew libraries:
  /Users/sjackman/.homebrew/opt/freetype/lib/libfreetype.6.dylib (freetype)
  /Users/sjackman/.homebrew/opt/libpng/lib/libpng16.16.dylib (libpng)```
```

**Comment by sjackman** [comment]
You can use…
```sh
brew deps --tree --include-build matplotlib
```
to troubleshoot the unexpected dependency.

**Comment by sjackman** [comment]
> matplotlib appears to depend on gcc now. Do you know why this change?

Never mind this question. TravisCI is on Ubuntu 14.

**Comment by sjackman** [comment]
There was previously a bottle built for El Capitan. This build failed on El Capitan with the error:
```
==> python -c import matplotlib
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/matplotlib/__init__.py", line 187, in <module>
    "Matplotlib requires six>=1.10; you have %s" % six.__version__)
ImportError: Matplotlib requires six>=1.10; you have 1.4.1
```
https://jenkins.brew.sh/job/Homebrew%20Science%20Pull%20Requests/1747/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/test_homebrew_science_matplotlib/

Does this error mean anything to you?

**Comment by davidchall** [comment]
Hi @sjackman 

Matplotlib should be finding the python package `six` version 1.10.0 because it is [listed as a resource](https://github.com/davidchall/homebrew-science/blob/72f0fb5e0b87ffd06cd94ab59341b44c0831e0e3/matplotlib.rb#L122) for this formula. Not sure what's happening here.

**Comment by sjackman** [comment]
```
❯❯❯ /usr/bin/python -c 'import six; print six.__version__'
1.4.1
```

It looks like macOS comes with `six` version 1.4.1, and it's being picked up before the resource version.

**Comment by davidchall** [comment]
Is this a problem with the Homebrew python environment used in testing? Or do I need to manually set the PYTHONPATH?

**Comment by sjackman** [comment]
I'm afraid that I don't know the answer to that question. It's odd that it fails on both Sierra and High Sierra, but fails on El Capitan. It failed not during installation but during `brew test homebrew/science/matplotlib`.

```
 Testing homebrew/science/matplotlib
/usr/bin/sandbox-exec -f /tmp/homebrew20171201-70314-1eo82kd.sb /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-science/matplotlib.rb --verbose
==> python -c import matplotlib
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/matplotlib/__init__.py", line 187, in <module>
    "Matplotlib requires six>=1.10; you have %s" % six.__version__)
ImportError: Matplotlib requires six>=1.10; you have 1.4.1
Error: homebrew/science/matplotlib: failed
```

@Homebrew/science Any idea why `brew test matplotlib` fails only on El Capitan but is fine on Sierra and High Sierra?

**Comment by sjackman** [comment]
I thought that perhaps it may be because Sierra and Higher Sierra include a newer version of `six`, but that does not seem to be the case. macOS 10.12 Sierra also includes `six` 1.4.1.

**Comment by MikeMcQuaid** [comment]
This tap has been deprecated and will shortly be archived.

If you wish to [migrate other widely used formulae](https://github.com/Homebrew/homebrew-science/issues/6331) to Homebrew/homebrew-core please [submit a pull request](http://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request.html) there.
